numpy/random/_generator.pyx

```diff
@@ -214,17 +214,17 @@ cdef class Generator:
 
     # Pickling support:
     def __getstate__(self):
-        return self.bit_generator.state
+        return self.bit_generator.__getstate__()
```

Comment:
Should be good, so don't think we need to iterate more (it is nicer than before!).  But thought I would comment that I would think we can just return the `bit_generator` here (which then gets pickled), so that `Generator` need not know that the bit generator even _has_ a getstate/state.

Reply:
I've removed this, simplified a bit more, and made sure it work with legacy pickles (<2 and <1.22, which had different `__reduce__` implementations).

I think it is ready now (subject to green). Only question remaining is whether to go ahead and return `type(self)` instead of `self.__class__.___name__`?